src/drivers/sdl/lv_sdl_window.c

```diff
@@ -208,20 +209,46 @@ static void flush_cb(lv_display_t * disp, const lv_area_t * area, uint8_t * px_m
 #if LV_USE_DRAW_SDL == 0
     lv_sdl_window_t * dsc = lv_display_get_driver_data(disp);
     if(LV_SDL_RENDER_MODE == LV_DISPLAY_RENDER_MODE_PARTIAL) {
-        int32_t y;
-        uint8_t * fb_tmp = dsc->fb_act;
+
+        uint8_t rotated_buf[32 * 1024];
+        lv_area_t rotated_area;
+        lv_display_rotation_t rotation = lv_display_get_rotation(disp);
+        if(rotation == LV_DISPLAY_ROTATION_90) {
+            int32_t w = lv_area_get_width(area);
+            int32_t h = lv_area_get_height(area);
+            lv_color_format_t cf = lv_display_get_color_format(disp);
+            lv_draw_sw_rotate(px_map, rotated_buf,
+                              w, h, lv_draw_buf_width_to_stride(w, cf),
```

Comment:
You're rotation `px_map` using the width and height of `area`. I think this only works if the area spans all of `px_map`, doesn't it? If the display is large, `flush_cb` will be called multiple times with different subareas that point into `px_map`. Maybe we could adapt `lv_draw_sw_rotate` to also accept an area argument so that only the pixels inside the area to be drawn are actually rotated?

Reply:
It should work in PARTIAL mode, as `px_map` has the dimension described by `area`. In DIRECT mode the whole rotation thing is not working like this because the not rotated area is already rendered into the middle of the frame buffer.

Reply:
Oh, maybe I saw this back when my still erroneously used direct rendering then, sorry. :facepalm: